### PR TITLE
Skip over empty archive batches instead of terminating the archive process

### PR DESCRIPTION
**Pull request by WilliamBZA**
Resolves https://github.com/Particular/ServicePulse/issues/592

Adds a test to reproduce the behavior and then proposes a fix which is to simply skip over empty archive batches so that the rest of the archive operation continues as it should instead of terminating.

### CODE REVIEW

File: src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        public void ArchiveGroup_skips_over_empty_batches_but_still_completes()
+        {
+            // Arrange
+            using (var documentStore = InMemoryStoreBuilder.GetInMemoryStore())
+            {
+                var groupId = "TestGroup";
+
+                using (var session = documentStore.OpenSession())
+                {
+                    var secondAchiveBatch = new ArchiveBatch { Id = ArchiveBatch.MakeId(groupId, ArchiveType.FailureGroup, 1) };
```
Comment (mikeminutillo) [inline src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs]: Would it make sense to pull this Id out as a `emptyBatchId` or something? It is used below as well.

File: src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    session.Store(secondAchiveBatch);
+
+                    var archiveOperation = new ArchiveOperation
```
Comment (mikeminutillo) [inline src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs]: Why is this needed? It feels like this test knows a lot about the inner workings of the archive subsystem.

Comment (WilliamBZA) [reply src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs]: It does, it's specifically a unit test intended to test a specific scenario. In order to set that scenario up it needs to know what the data looks like.

File: src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs (hunk 1 above)
Comment (WilliamBZA) [reply src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs]: 👍

**Comment by WilliamBZA** [comment]
By calling the `ArchiveAllInGroup` method before setting this code up, it won't get into the state that the test is trying to validate.

The specific scenario is that an archive group operation was started and stopped but some of the data was not cleaned up which means that when a second call to archive the same group is run, it terminates early because it thinks the archive operation is completed. In order to test this, this test first introduces the data that simulates the "started and stopped but not cleaned up" and then calls the `ArchiveAllInGroup` method again, thereby testing what happens when the user invokes the archive operation on the same group a second time.

**Comment by mikeminutillo** [comment]
@WilliamBZA Would it make sense to rename the variables to `oldArchiveBatch` and `oldArchiveOperation` then? Just to make it more explicit what is going on.

**Comment by WilliamBZA** [comment]
@mikeminutillo I've renamed to `previous`
